src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -254,287 +169,226 @@ inline InferenceEngine::Blob::Ptr fp32_to_precision_blob(InferenceEngine::Blob::
         }
+
         memset(bias->buffer(), 0, bias->byteSize());
 
         return bias;
     };
-    auto intBiases = biasMaker();
 
+    const auto intBiases = biasMaker();
+    const auto& dstScale = InferenceEngine::getInjectedData<QuantizedLayerParams>(*wl)->_dst_quant.GetScale();
+
+    BiasQuantizationCallback<BiasesType> {common_data,
+        wl->_biases ? wl->_biases->buffer().as<float*>() : nullptr,
+        intBiases ? intBiases->buffer() : static_cast<BiasesType*>(nullptr),
+        dstScale
+    }.quantize();
+
+    wl->_biases = intBiases;
+}
+
+static std::map<InferenceEngine::Precision, void (*)(InferenceEngine::WeightableLayer*, QuantizationCallbackData&)> weightsPrecisions {
+    {
+        InferenceEngine::Precision::I8, quantizeWeights<int8_t>
+    },
+    {
+        InferenceEngine::Precision::I16, quantizeWeights<int16_t>
+    }
+};
+
+static std::map<InferenceEngine::Precision, void (*)(InferenceEngine::WeightableLayer*, QuantizationCallbackData&)> biasesPrecisions {
+    {
+        InferenceEngine::Precision::I8, quantizeBiases<int8_t>
+    },
+    {
+        InferenceEngine::Precision::I16, quantizeBiases<int16_t>
+    },
+    {
+        InferenceEngine::Precision::I32, quantizeBiases<int32_t>
+    },
+    {
+        InferenceEngine::Precision::fromType<gna_rich_bias_t>(), quantizeBiases<gna_rich_bias_t>
+    }
+};
+
+static void quantizeWeightsBiases(InferenceEngine::WeightableLayer *wl) {
     float input_scale_factor = 1.f;
-    if (InferenceEngine::CNNNetHasPrevLayer(conv)) {
+
+    if (InferenceEngine::CNNNetHasPrevLayer(wl)) {
         auto quantDataForInputLayer =
-            InferenceEngine::getInjectedData<QuantizedLayerParams>(*InferenceEngine::CNNNetPrevLayer(conv).get());
+            InferenceEngine::getInjectedData<QuantizedLayerParams>(*InferenceEngine::CNNNetPrevLayer(wl).get());
         input_scale_factor = quantDataForInputLayer->_dst_quant.GetScale();
         if (std::isnan(input_scale_factor) ||
             std::isinf(input_scale_factor)) {
             IE_THROW() << "Unsupported input scale factor value " << input_scale_factor;
         }
     }
-    if (conv->outData[0]->getDims().size() < 2) {
-        IE_THROW() << "Unsupported output dims size for " << conv->name <<", should be > 1, but " << conv->outData[0]->getDims().size();
-    }
-    if (conv->insData[0].lock().get()->getDims().size() < 2) {
-        IE_THROW() << "Unsupported input dims size for " << conv->name << ", should be > 1, but " << conv->insData[0].lock().get()->getDims().size();
-    }
-    auto inputData = conv->insData[0].lock();
 
-    uint32_t num_rows = getBiasSizeForLayer(conv);
-    if (num_rows == 0) {
-        THROW_GNA_EXCEPTION << "Invalid num rows";
-    }
-    uint32_t num_columns = conv->_weights->size() / num_rows;
+    size_t num_rows = 0;
+    size_t num_columns = 0;
 
-    uint32_t num_rows_padded = num_rows;
-    uint32_t num_columns_padded = num_columns;
+    getNumRowsColumns(wl, num_rows, num_columns);
```

Comment:
Detail: If `getNumRowsColumns` returns pair or tuple then
```cpp
size_t num_rows, num_columns;
std::tie(num_rows, num_columns) = getNumRowsColumns(wl);
```

rows, colums will be set once instead set and overwrite.